public/pages/workflow_detail/workflow_inputs/input_fields/map_array_field.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiAccordion,
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLink,
+  EuiPanel,
+  EuiText,
+} from '@elastic/eui';
+import { Field, FieldProps, getIn, useFormikContext } from 'formik';
+import {
+  IConfigField,
+  MapArrayFormValue,
+  MapEntry,
+  WorkflowFormValues,
+} from '../../../../../common';
+import { MapField } from './map_field';
+
+interface MapArrayFieldProps {
+  field: IConfigField;
+  fieldPath: string; // the full path in string-form to the field (e.g., 'ingest.enrich.processors.text_embedding_processor.inputField')
+  label?: string;
+  helpLink?: string;
+  helpText?: string;
+  keyPlaceholder?: string;
+  valuePlaceholder?: string;
+  onFormChange: () => void;
+  onMapAdd?: (curArray: MapArrayFormValue) => void;
+  onMapDelete?: (idxToDelete: number) => void;
+}
+
+/**
+ * Input component for configuring an array of field mappings
+ */
+export function MapArrayField(props: MapArrayFieldProps) {
```

Comment:
nit: maybe consider to wrap it with memo to ensure that the component only re-renders when its props change

Reply:
Good point! Since this is still in rapid development, I'm going to leave as-is for now, and will consider that a standalone task to audit all components before release. Created a tracking issue and triaged for 2.17: #247